src/Microsoft.DocAsCode.Build.Common/ValidateBookmark.cs

```diff
@@ -19,48 +19,57 @@ public class ValidateBookmark : HtmlDocumentHandler
+            _linksWithBookmark[outputFile] =
+                (from node in GetNodesWithAttribute(document, "href")
+                 let link = node.GetAttributeValue("href", null)
                  let index = link.IndexOf("#")
                  where index != -1 && PathUtility.IsRelativePath(link)
-                 select Tuple.Create(HttpUtility.UrlDecode(link.Remove(index)), link.Substring(index + 1)) into pair
-                 where !WhiteList.Contains(pair.Item2)
-                 select pair).ToList();
+                 select new LinkItem { Href = HttpUtility.UrlDecode(link.Remove(index)), Bookmark = link.Substring(index + 1), SourceLineNumber = node.GetAttributeValue("sourceStartLineNumber", 0), TargetLineNumber = node.Line } into item
+                 where !WhiteList.Contains(item.Bookmark)
+                 select item).ToList();
             var anchors = GetNodeAttribute(document, "id").Concat(GetNodeAttribute(document, "name"));
             _registeredBookmarks[outputFile] = new HashSet<string>(anchors);
         }
 
         public override Manifest PostHandle(Manifest manifest)
         {
-            foreach (var item in _bookmarks)
+            foreach (var pair in _linksWithBookmark)
             {
-                string path = item.Key;
-                foreach (var b in item.Value)
+                string currentFile = pair.Key;
+                foreach (var linkItem in pair.Value)
                 {
-                    string linkedToFile = b.Item1 == string.Empty ? path : b.Item1;
-                    string anchor = b.Item2;
-                    HashSet<string> anchors;
-                    if (_registeredBookmarks.TryGetValue(linkedToFile, out anchors) && !anchors.Contains(anchor))
+                    string linkedToFile = linkItem.Href == string.Empty ? currentFile : linkItem.Href;
+                    string bookmark = linkItem.Bookmark;
+                    HashSet<string> bookmarks;
+                    if (_registeredBookmarks.TryGetValue(linkedToFile, out bookmarks) && !bookmarks.Contains(bookmark))
                     {
-                        string currentFileSrc = _fileMapping[path];
+                        string currentFileSrc = _fileMapping[currentFile];
                         string linkedToFileSrc = _fileMapping[linkedToFile];
-                        Logger.LogWarning($"Output file {path} which is built from src file {currentFileSrc} contains illegal link {linkedToFile}#{anchor}: the file {linkedToFile} which is built from src {linkedToFileSrc} doesn't contain a bookmark named {anchor}.", file: path);
+                        if (linkItem.SourceLineNumber == 0)
+                        {
+                            Logger.LogWarning($"{currentFile} contains illegal link: {linkItem.Href}#{bookmark}. The file {linkedToFile} doesn't contain a bookmark named {bookmark}, please check the template you applied.", file: currentFile, line: linkItem.TargetLineNumber.ToString());
```

Comment:
>please check the template you applied [](start = 182, length = 37)

Is invalid bookmark always from template here? e.g. directly writting HTML in markdown.